### DISCUSSION TITLE
Whats the point of this?

### DISCUSSION THREAD

**Discussion by Killerdrax** [medspacy/medspacy#306] [Q&A]
I mean I tried using it without the target rules and it isnt showing any entities. If I have to specify each and every entity for each input why this even exists rather than just using if functions. I am not able to get any entities unless I give target rules.

**Comment by burgersmoke** [comment]
You raise a good question @Killerdrax .  Medspacy is not meant to be an out-of-the-box one size fits all pipeline to immediately apply into production.

Your are not the only person who asks questions like these.  This is one of the reasons we call medspacy a toolkit.  It has pieces that you can put together with other spacy components from other offerings. If I were to highlight some of the major benefits of why people use medspacy they would include:

- Section Detection
- ConText for modifying entity attributes (e.g., negation, historical, etc)
- Ability to write custom entity rules (I think this is perhaps mainly what you raise as a downside)

While these are some of the benefits from some users, your question likely remains "well how do I get clinical concepts and entities?"  There are several options available to you since this is a toolkit where you can do a mix and match of components:

- One of these options is to implement [UMLS entity extraction with the EntityLinker in scispacy](https://github.com/allenai/scispacy).
- Another is to use QuickUMLS provided by medspacy.  Note that as a medspacy team, we are not allowed by licensing agreement to redistribute UMLS resources.  So instead we provide guidance on how users can download them on their own and then [turn these into a QuickUMLS database](https://github.com/Georgetown-IR-Lab/QuickUMLS?tab=readme-ov-file#installation).
- Finally you can also set up your own custom entities.  This can be useful where knowledge for these entities does not yet exist.  This was one of our [first use cases of medspacy in early 2020](https://github.com/abchapman93/VA_COVID-19_NLP_BSV) as the COVID-19 pandemic was emerging, and before we had names like 'COVID-19', 'SARS-COV-2', etc.

Does this answer your question @Killerdrax ?
